### PR TITLE
feat: implement first version

### PR DESCRIPTION
**Pull request by patrickhoogkamer**

### CODE REVIEW

File: src/Model.php [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        return Str::of((new ReflectionClass($this))->getShortName())
+            ->lower()
+            ->plural()
+            ->prepend('/')
+            ->toString();
+    }
+
+    public function perPage(): int
+    {
+        return $this->perPage;
```
Comment (tomasvanrijsse) [inline src/Model.php:36]: 🐮 we could just do it like this and remove one more property.

```suggestion
        return 30;
```

File: src/Model.php [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    public static function find(mixed $id): ?static
+    {
+        return static::query()->find($id);
+    }
+
+    /**
+     * @param  array<string, mixed>  $item
+     * @param  array<int, mixed>|null  $included
+     */
+    public function hydrate(array $item, array|null $included = null): static
```
Comment (tomasvanrijsse) [inline src/Model.php:68]: I think it would be nice to extract this method

**Review by tomasvanrijsse — APPROVED** [review]
The two comments are cowish, so I'll leave it up to you if you want to implement them now,later or never.

File: src/Model.php (hunk 2 above)
Comment (patrickhoogkamer) [reply src/Model.php:68]: Will do in a new issue.

File: src/Model.php (hunk 1 above)
Comment (patrickhoogkamer) [reply src/Model.php:36]: You can extend with the method or the property now. Only the method is fine, but I'm not sure if it makes a real difference. For now I'm merging it so we can use it faster.

**Pull request merged**